File: clients/python/SampleMatchMakerClient.py
import socket
import random
import time
import sys
from signal import signal, SIGPIPE, SIG_DFL  
signal(SIGPIPE,SIG_DFL)

clientsocket = socket.socket()

def generate_candidates(i, N, score):
  
  '''
  TO DO: Change candidates to your own algorithm
  DON'T FORGET THE \n
  '''
  if i < 30:
    candidates = "0." + ":0.".join(str(e)
                                   for e in [str(int(random.random()*10000))
                                             for i in range(N)])+"\n"
  else:                                       
    candidates = "1" + ":0"*(N-1) + "\n"
  return candidates

if __name__ == '__main__':
  clientsocket.connect(('localhost', 20001))
  socketfile = clientsocket.makefile('rw')
  socketfile.write('Matchmaker\n')
  socketfile.flush()
  # Init
  N = 0
    
  # Start Read from the 20 Random Candidates and scores from Server (First Round)
  for i in range(20):
    inputLine = socketfile.readline()
    print("Server says:" + inputLine)
    tmp = inputLine.split(":")
    # Check N and set
    if i == 0:
      # Set
      N = len(tmp) - 1
      if (N < 2 or N > 200):
        print("invalid N")
        sys.exit()
    candidates = tmp[:-1]
    score = tmp[-1]
    '''
    TO DO: You could use the candidates and score to implement algorithm
    '''

  # Start Write Candidates to P and Read from the Scores (Next Rounds)
  for i in range(20, 40):
    candidates = generate_candidates(i, N, score)
    socketfile.write(candidates)
    socketfile.flush() 
    inputline = socketfile.readline()
    print("Server says:" + inputline)
    if inputline == "END\n":
      sys.exit()
    score = inputline.split(":")[1]
    time.sleep(.2)
 
  print("Server says:" + socketfile.readline())
  clientsocket.close()

File: clients/java/SampleMatchMakerClient.java
import java.io.*;
import java.net.*;
import java.util.*;

public class SampleMatchMakerClient {

  //  TO DO: Change candidates to your own algorithm
  private static String generate_candidates(int i, int N, String score){
    Random generator = new Random();
    String candidate = "";
    if (i < 30){
      for(int k = 0; k < N; k++) {
        if (k>0)
          candidate = candidate + ":";
          candidate = candidate + "0." + (Integer.toString((int)(generator.nextDouble() * 10000)));
      }
    }
    else {
      candidate ="1";
      for(int k = 0; k < N-1; k++)
        candidate += ":0";
    }
    return candidate;
  }

  public static void main(String[] args) throws Exception {
    Socket socket = null;
    PrintWriter out = null;
    BufferedReader in = null;
    try {
      socket = new Socket("localhost", 20001);
      out = new PrintWriter(socket.getOutputStream(), true);
      in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
    } 
    catch (UnknownHostException e) {
      System.out.println("Unknown host");
      System.exit(-1);
    } 
    catch (IOException e) {
      System.out.println("No I/O");
      System.exit(-1);
    }
    out.println("Matchmaker");    
    // INIT
    int N = 0;
    String score = "";
    // Start Read from the 20 Random Candidates and scores from Server (First Round)
    for (int i = 0; i < 20; i++) {
      String inputLine = in.readLine();
      System.out.println("Server says: " + inputLine);
      String [] tmp = inputLine.split(":");
      // Check N and set
      if (i == 0){
        N = tmp.length - 1;
        if(N < 2 || N > 200) {
          System.out.println("invalid N");
          System.exit(-1);
        }
      }
      String [] candidates = Arrays.copyOf(tmp, tmp.length-1);
      score = tmp[tmp.length-1];
      // TO DO: You could use the candidates and score to implement algorithm

    }
    for (int i = 20; i < 40; i++) {
      String candidates = generate_candidates(i, N, score);
      out.println(candidates);
      String inputLine = in.readLine();
      System.out.println("Server says: " + inputLine);
      if (inputLine.equals("END")){
       System.exit(-1);
      }
      String [] tmp = inputLine.split(":");
      score = tmp[1];
      Thread.sleep(200);
    }

    String inputLine = in.readLine();
    System.out.println("Server says: " + inputLine);    
    out.close();
    in.close();
    socket.close();
  }
}